packages/layerchart/src/lib/components/ForceSimulation.svelte

```diff
@@ -1,67 +1,335 @@
-  import { getContext } from 'svelte';
+  import { createEventDispatcher, getContext } from 'svelte';
+
   import { forceSimulation, type Force } from 'd3-force';
 
   const { data } = getContext('LayerCake');
 
-  export let forces: Record<string, Force<any, any>>;
+  const dispatch = createEventDispatcher<{
+    start: null;
+    tick: { alpha: number; alphaTarget: number };
+    change: null;
+    end: null;
+  }>();
+
+  // MARK: Public Props
 
-  export let alpha = 1;
-  export let alphaTarget = 0;
-  export let alphaDecay = 1 - Math.pow(0.001, 1 / 300);
-  export let alphaMin = 0.001;
+  type Forces = Record<string, Force<any, any>>;
+  export let forces: Forces;
+
+  export let alpha: number = 1;
+  export let alphaTarget: number = 0;
+  export let alphaDecay: number = 1 - Math.pow(0.001, 1 / 300);
+  export let alphaMin: number = 0.001;
 
   export let velocityDecay = 0.4;
 
-  /** Clone data since simulation mutates original */
-  export let cloneData = false;
+  type AlphaResponse = (alpha: number, alphaTarget: number) => number;
+  export let alphaResponse: AlphaResponse | undefined = undefined;
+
+  export let stopped = false;
 
   let _static = false;
   /** If true, will only update nodes after simulation has completed */
   export { _static as static };
 
-  let simulation = forceSimulation(cloneData ? structuredClone($data) : $data);
+  /** Clone data since simulation mutates original */
+  export const cloneData = false;
+
+  // MARK: Private Props
+
+  let nodes: any[] = [];
+
+  const simulation = forceSimulation().stop(); //.nodes(nodesFromData());
+
+  // Only dynamic simulations distinguish between paused and running state.
+  // Invariant: Static simulations always keep `paused = true`.
+  let paused: boolean = true;
+
+  // MARK: Reactivity Triggers
+
+  $: {
+    // Any time the `stopped` prop gets toggled we
+    // update the running state of the simulation:
+    updateStateOnChangeOf({
+      stopped,
+      _static,
+    });
+  }
+
+  $: {
+    // Any time the `static` prop gets toggled we
+    // either attach or detach our internal event listeners:
+    updateEventListenersOnChangeOf({ _static });
+  }
+
+  $: {
+    // Any time the `$data` store gets changed we
+    // pass them to the internal d3 simulation object:
+    updateNodesOnChangeOf({ data: $data });
+  }
+
+  $: {
+    // Any time the `forces` prop gets changed we
+    // pass them to the internal d3 simulation object:
+    updateForcesOnChangeOf({ forces });
+  }
+
+  $: {
+    // Any time the `alpha` prop gets changed we
+    // pass it to the internal d3 simulation object:
+    updateAlphaOnChangeOf({ alpha });
+  }
 
   $: {
+    // Any time any of the the alpha props get changed we
+    // pass them all to the internal d3 simulation object
+    // (they are cheap, so passing them as a batch is fine!):
+    updateSettingsOnChangeOf({
+      alphaTarget,
+      alphaMin,
+      alphaDecay,
+      velocityDecay,
+      _static,
+    });
+  }
```

Comment:
Any reason we couldn't move each of these function implementations inline to the related `$: { ... }` statements.  I don't think they are needed to hide reactivity from Svelte (sometimes a trick I use, especially for circular references).  I think this would help readability.  I spot checked most of them and they looked to be called once. 

We could also add a `runStaticOrResumeDynamicSimulation()` to simplify them as well.

```js
function runStaticOrResumeDynamicSimulation() {
    if (_static) {
      runStaticSimulation();
    } else {
      resumeDynamicSimulation(alpha);
    }
}
```

Reply:
> […] I don't think they are needed to hide reactivity from Svelte (sometimes a trick I use, especially for circular references). […]

That's actually exactly why I added them at some point in the refactor. 😄
But on second glance it looks like in the state they ended up in no hiding is necessary anymore, so I'll try to inline them. 👍🏻